src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/KeyController.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                Keys = keys.Select(key => new CryptoKeyDto
+                {
+                    Name = key.Name,
+                    Id = key.Id.ToString()
+                })
+            };
+        }
+
+        /// <summary>
+        ///   Create a new key.
+        /// </summary>
+        /// <param name="arg">
+        ///   The name of the key.
+        /// </param>
+        /// <param name="type">
+        ///   "rsa"
+        /// </param>
+        /// <param name="size">
+        ///   The key size in bits, if the type requires it.
+        /// </param>
+        [HttpGet, HttpPost, Route("key/gen")]
+        public async Task<CryptoKeyDto> Create(string arg,
+            string type,
+            int size)
+        {
+            if (String.IsNullOrWhiteSpace(arg))
+                throw new ArgumentNullException(nameof(arg), "The key name is required.");
+            if (String.IsNullOrWhiteSpace(type))
+                throw new ArgumentNullException(nameof(type), "The key type is required.");
+
+            var key = await IpfsCore.KeyApi.CreateAsync(arg, type, size, Cancel);
+            return new CryptoKeyDto
+            {
+                Name = key.Name,
+                Id = key.Id.ToString()
+            };
+        }
+
+        /// <summary>
+        ///   Remove a key.
+        /// </summary>
+        /// <param name="arg">
+        ///   The name of the key.
+        /// </param>
+        [HttpGet, HttpPost, Route("key/rm")]
+        public async Task<CryptoKeysDto> Remove(string arg)
+        {
+            if (String.IsNullOrWhiteSpace(arg))
+                throw new ArgumentNullException(nameof(arg), "The key name is required.");
+
+            var key = await IpfsCore.KeyApi.RemoveAsync(arg, Cancel);
+            var dto = new CryptoKeysDto();
+            if (key != null)
+            {
+                dto.Keys = new[]
+                {
+                    new CryptoKeyDto
+                    {
+                        Name = key.Name,
+                        Id = key.Id.ToString()
+                    }
+                };
+            }
+
+            return dto;
+        }
+
+        /// <summary>
+        ///   Rename a key.
+        /// </summary>
+        /// <param name="arg">
+        ///   The old and new key name.
+        /// </param>
+        [HttpGet, HttpPost, Route("key/rename")]
+        public async Task<CryptoKeyRenameDto> Rename(string[] arg)
+        {
+            if (arg.Length != 2)
+                throw new ArgumentException("Missing the old and/or new key name.");
+
+            var key = await IpfsCore.KeyApi.RenameAsync(arg[0], arg[1], Cancel);
+            var dto = new CryptoKeyRenameDto
+            {
+                Was = arg[0],
+                Now = arg[1],
+                Id = key.Id.ToString()
+
+                // TODO: Overwrite
+            };
+            return dto;
+        }
+
+        // TODO: import
+        // TODO: export
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)